flink-autoscaler-plugin-jdbc/src/main/java/org/apache/flink/autoscaler/jdbc/event/JdbcEventInteractor.java

```diff
@@ -152,4 +165,49 @@ protected List<AutoScalerEvent> queryEvents(String jobKey, String reason) throws
     void setClock(@Nonnull Clock clock) {
         this.clock = Preconditions.checkNotNull(clock);
     }
+
+    @Nullable
+    ExpiredEventsResult queryExpiredEventsAndMaxId(Duration eventHandlerTtl) throws Exception {
+        var query =
+                "SELECT COUNT(1) records_num, max(id) max_target_id "
+                        + "FROM t_flink_autoscaler_event_handler "
+                        + "WHERE create_time < ? AND id < ("
+                        + "   SELECT id FROM t_flink_autoscaler_event_handler "
+                        + "   WHERE create_time >= ? ORDER BY id ASC LIMIT 1)";
+        var date = Timestamp.from(clock.instant().minusMillis(eventHandlerTtl.toMillis()));
+        try (var pstmt = conn.prepareStatement(query)) {
+            pstmt.setObject(1, date);
+            pstmt.setObject(2, date);
+            ResultSet resultSet = pstmt.executeQuery();
+            if (!resultSet.next()) {
+                return null;
+            }
+            var result = new ExpiredEventsResult(resultSet.getInt(1), resultSet.getLong(2));
+            resultSet.close();
```

Comment:
Don't need it, copy from the comment of `java.sql.ResultSet#close`. Let us keep all queries are same style.

`Note: A ResultSet object is automatically closed by the Statement object that generated it when that Statement object is closed, re-executed, or is used to retrieve the next result from a sequence of multiple results.`